src/tensor_ops/conv2d/cuda_kernel.rs

```diff
@@ -68,8 +63,31 @@ where
         rhs: &Tensor<R, E, Self>,
         out: &mut Tensor<O, E, Self>,
     ) -> Result<(), Self::Err> {
-        if !self.dev.has_func(Self::MOD, Self::FNS[0]) {
-            self.dev.load_ptx(PTX_SRC.into(), Self::MOD, Self::FNS)?;
+        let name = mod_name(&op);
+        if !self.dev.has_func(&name, "unfold_input") {
+            let src = KERNEL_SRC;
+            let src = src.replace("op.stride", &op.stride.to_string());
+            let src = src.replace("op.padding", &op.padding.to_string());
+            let src = src.replace("op.kernel", &op.kernel.to_string());
+            let src = src.replace("op.batch", &op.batch.to_string());
+            let src = src.replace("op.chan_in", &op.chan_in.to_string());
+            let src = src.replace("op.chan_out", &op.chan_out.to_string());
+            let src = src.replace("op.h_in", &op.h_in.to_string());
+            let src = src.replace("op.h_out", &op.h_out.to_string());
+            let src = src.replace("op.w_in", &op.w_in.to_string());
+            let src = src.replace("op.w_out", &op.w_out.to_string());
+            let src = src.replace("$TY", Self::TYPENAME);
+            let ptx = compile_ptx(src).unwrap();
```

Comment:
@nkoppel thoughts on this? This makes the first pass through the network take a bit longer since it has to compile the PTX, but:
1. Its less kernels to compile at compile time
2. we can inline values

I'm just fooling around with this for now (it didn't make that much of a performance impact for conv2d), but curious what others might think of JITing kernels

Reply:
This seems like a pretty good idea, especially considering that conv2d is the kind of operation that is likely to be run several times with the exact same parameters. However, I think that we should generally avoid making shapes a part of kernels' compile time configuration, so that we can guarantee a finite number of kernels are created.

I would also be interested to know if there is benefit to caching kernels in files over recompiling them every run.